### PR TITLE
[SPARK-15562][ML] Delete temp directory after program exit in DataFrameExample

### PR DESCRIPTION
**Pull request by dding3**
## What changes were proposed in this pull request?

Temp directory used to save records is not deleted after program exit in DataFrameExample. Although it called deleteOnExit, it doesn't work as the directory is not empty. Similar things happend in ContextCleanerSuite. Update the code to make sure temp directory is deleted after program exit.
## How was this patch tested?

unit tests and local build.

### CODE REVIEW

**Comment by srowen** [comment]
I'm not sure this works, since the checkpoint directory should not exist before the job starts? I could be misremembering but I think this may be on purpose. At least, the temp dir needs to be deleted after being created.

**Comment by dding3** [comment]
Thanks for pointing out it. I think you are right, delete the checkpoint directory after it's created is on purpose. I will add "delete checkpoint directory" back. Besides, the temp directory will be deleted after the test finished as it has been registered to delete if VM shuts down when it's created. I have verified that, the temp checkpoint directory is not deleted after test finished in the previous code, while it's deleted in the new code.

File: core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala [hunk 1]
```diff
@@ -32,7 +32,7 @@ import org.apache.spark.internal.Logging
 import org.apache.spark.rdd.{RDD, ReliableRDDCheckpointData}
 import org.apache.spark.shuffle.sort.SortShuffleManager
 import org.apache.spark.storage._
-
+import org.apache.spark.util.Utils
 /**
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala]: Nit: space after this import. Otherwise I think this is OK, and an improvement.

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by dding3** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**